src/main/java/com/minecolonies/coremod/colony/Structures.java

```diff
@@ -25,23 +27,32 @@
     /**
      * Ignore the styles from the generic folder.
      */
-    private static final String NULL_STYLE = "schematics";
+    private static final String NULL_STYLE = "";
 
     /**
      * Ignore the styles from the miner folder.
      */
     private static final String MINER_STYLE = "miner";
 
-    private static final String                    SCHEMATIC_EXTENSION   = ".nbt";
-    private static final String                    SCHEMATICS_ASSET_PATH = "/assets/minecolonies/schematics/";
+    private static final String                    SCHEMATIC_EXTENSION        = ".nbt";
+    private static final String                    SCHEMATICS_ASSET_PATH      = "/assets/minecolonies/schematics/";
+    public  static final String                    SCHEMATICS_HUTS            = "huts";
+    public  static final String                    SCHEMATICS_DECORATIONS     = "decorations";
     //Hut, Levels
     @NotNull
-    private static final Map<String, Integer>      hutLevelsMap          = new HashMap<>();
+    private static       Map<String, Integer>      hutLevelsMap          = new HashMap<>();
     //Hut, Styles
     private static       Map<String, List<String>> hutStyleMap           = new HashMap<>();
     //Decoration, Style
     private static       Map<String, List<String>> decorationStyleMap    = new HashMap<>();
 
+    /* md5 hash for the schematics
+     * format is:
+     * huts/stone/builder1.nbt -> hash
+     * decorations/decoration/Well.nbt -> hash
+     */
+    private static       Map<String, String>       md5Map                = new HashMap<>();
```

Comment:
> The idea behind you using these hashes is to find the differences between two files. So two different files (regardless of the fact that they belong to the same building, even if it is rotated) should never and i mean never have the same hash.

No the idea is to make sure the client use the correct schematic.

> That would complete negate the principle of what you are working here.
No

>On top of that, if you had two files with the same hash, your method to get a structure name from a hash would only return the first one. Which is maybe not the one you requested.

One is better than none.

I use the hash to fix Work Order where the schematic still exist but have been renamed or moved. Yes this is not full proof but it is better that failing a WorkOrder and cancelling it. Or have a worthless building just because the schematic moved or its style name have been changed